test/nodes/test_shaper.py

```diff
@@ -0,0 +1,563 @@
+from haystack import Pipeline, Document
+
+
+def test_basic_function_invocation(tmp_path):
```

Comment:
Is it possible to define a `Shaper` with only using python and not having to rely on the yaml definition? If so could we also provide an example of how that would look?

Reply:
Hmm, interesting perspective. We made this component so we can do manipulations of the invocation context by using yaml definitions only. One could define these directives via Shaper config dict if needed. That would be enough, right?

Reply:
> Hmm, interesting perspective. We made this component so we can do manipulations of the invocation context by using yaml definitions only.

Ahh okay, I see. I think we still want it to be usable natively in python since we also encourage developers to use Jupyter notebooks for the development of pipelines where we also show how the pipelines can be defined using the Pipeline components directly (instead of using yaml). But I could be wrong about this, what do you think @mathislucka ?

> One could define these directives via Shaper config dict if needed. That would be enough, right?

I think so. It would be good to have an example in here just to see what that would look like.

Reply:
See unit test in https://github.com/deepset-ai/haystack/pull/3815/commits/8fa8bdbed3a6878d1ca676c5da099857c5700bc4